### PR TITLE
Add new chart test workflow

### LINKED ISSUE #772: Add tests for the chart installation and upgrade

### PR DESCRIPTION
**Pull request by FushuWang**
Fixes #772 

### Motivation

Add new chart test workflow.

### Modifications

Add new action

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Documentation

Check the box below.

Need to update docs? 

- [x] `no-need-doc` 
  
  (Please explain why)

### CODE REVIEW

**Review by maxsxu — COMMENTED** [review]
Seems the CI lint errors indicate that we should upgrade the Chart version?

**Comment by FushuWang** [comment]
> Seems the CI lint errors indicate that we should upgrade the Chart version?

Yeah, chart-testing will need a version bump when the chart has new changes. I updated a new version in the testing to trigger the test, So I just revert the chart version after the tests are working.

**Review by maxsxu — COMMENTED** [review]
@FushuWang Please resolve the conflicts

File: .github/workflows/chart_test.yml [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+on:
+  push:
+    paths:
+      - '.github/workflows/chart_test.yml'
+  pull_request:
+    branches:
+      - '*'
+    paths:
+      - 'charts/**'
+      - '.github/workflows/**'
```
Comment (maxsxu) [inline .github/workflows/chart_test.yml:30]: Should we restrict the paths to sn-platform?

**Pull request closed without merging**